File: backend-python/app/api_response.py
import time
from typing import Any, Dict, Optional


def _now_millis_str() -> str:
    """返回当前时间的毫秒时间戳字符串，兼容前端 `Number(res.timestamp)`。"""
    return str(int(time.time() * 1000))


def ok(data: Any) -> Dict[str, Any]:
    """成功响应包装，与 Go 版 APIResponse 结构保持一致。"""
    return {
        "code": "200",
        "data": data,
        "msg": "操作成功",
        "success": True,
        "timestamp": _now_millis_str(),
    }


def fail(code: str, msg: str, data: Optional[Any] = None) -> Dict[str, Any]:
    """失败响应包装，HTTP 层仍返回 200 状态码。"""
    return {
        "code": code,
        "data": data,
        "msg": msg,
        "success": False,
        "timestamp": _now_millis_str(),
    }

File: backend-node/src/shared/api-response/api-response.ts
/**
 * 统一 API 响应结构，与前端 ApiRes<T>、Go/Python 版 APIResponse 完全对齐。
 */
export interface ApiResponse<T> {
  code: string;
  data: T;
  msg: string;
  success: boolean;
  timestamp: string;
}

/**
 * 返回当前时间戳（毫秒）的字符串表示，前端使用 Number(timestamp)。
 */
function nowString(): string {
  return Date.now().toString();
}

/**
 * 成功响应包装。
 */
export function ok<T>(data: T): ApiResponse<T> {
  return {
    code: '200',
    data,
    msg: '操作成功',
    success: true,
    timestamp: nowString(),
  };
}

/**
 * 失败响应包装。
 */
export function fail(code: string, msg: string): ApiResponse<null> {
  return {
    code,
    data: null,
    msg,
    success: false,
    timestamp: nowString(),
  };
}

